src/main/java/jenkins/branch/BranchNameContributor.java

```diff
@@ -61,6 +62,11 @@ public void buildEnvironmentFor(Job j, EnvVars envs, TaskListener listener) thro
                 // Note: not using Branch.name, since in the future that could be something different
                 // than SCMHead.name, which is what we really want here.
                 envs.put("BRANCH_NAME", head.getName());
+                if (branch.getAction(PrimaryInstanceMetadataAction.class) != null) {
+                    envs.put("BRANCH_IS_PRIMARY", "true");
+                } else {
+                    envs.put("BRANCH_IS_PRIMARY", "false");
```

Comment:
Late review/question—instead of setting it to `"false"` (which is a truthy string), would it be possible to set it to the empty string?

```suggestion
                    envs.put("BRANCH_IS_PRIMARY", "");
```

This would make it easier for scripts to check this value without having to remember to convert it to a Boolean or comparing string literals. Please let me know if this makes sense, and I can open a PR.

(Hopefully it's not too late/disruptive to make this change since it looks like this hasn't been released yet. 🤞 )

Reply:
@usmonster 
If we haven't released since this was merged, I'd still be open to changing the behavior but I'm not convinced we should.  I generally think these variables being used by any number of systems, so designing specifically for shell script behavior isn't my goal.  `false` is much clearer in my java-centric mind.  Perhaps you could file a PR with the proposed change  some examples and tag @cwholmes and we can discuss in a thread that isn't off the tail end of a merged PR.  😄

Reply:
Thanks for your quick response, @bitwiseman!

To be clear, this isn't a shell script-specific proposal, but I was thinking more about the most common context in which these variables will be used, i.e. in the Groovy scripts that define Jenkins pipelines. Maybe the variable shouldn't even be set if it's not a primary branch? In any case, I'll file a new PR to formally propose and discuss this. Thanks again!

Reply:
my 2 cents is you should only be comparing for the string == "true".

imo current behaviour is fine. but 🤷

Reply:
👉  #246